CMakeLists.txt

```diff
@@ -118,81 +59,71 @@ else()
 	message( FATAL_ERROR "Boost library is not installed in your system" )    
 endif()
 
-
-# Create header file with the XX_ENABLE flags
-#set (HEADER_FILE "include/enable_flags.hpp")
-#file (WRITE ${HEADER_FILE} "//Enable Flags")
-
 # TBB Library 
-option( TBB_ENABLE "Require TBB library" ON )
-message( STATUS "TBB: " ${TBB_ENABLE} )
-if( TBB_ENABLE )
+option( GRPPI_TBB_ENABLE "Require TBB library" ON )
+message( STATUS "TBB: " ${GRPPI_TBB_ENABLE} )
+if( GRPPI_TBB_ENABLE )
     include_directories(${TBB_INCLUDE_DIRS} )
     find_package(TBB REQUIRED)
     add_definitions( "-DGRPPI_TBB" )
-#    file ( APPEND ${HEADER_FILE} "\n#define TBB_ENABLE ON")
-endif( TBB_ENABLE )
+endif( GRPPI_TBB_ENABLE )
 
 # Threads Library
 find_package(Threads REQUIRED)
```

Comment:
If we use this, we should consider taking advantage of this for the link of executables.

find_package( Threads )

In any executable do:
add_executable( test test.cpp )
target_link_libraries( test ${CMAKE_THREAD_LIBS_INIT} )